addons/mail/models/ir_attachment.py

```diff
@@ -36,6 +36,16 @@ def register_as_main_attachment(self, force=True):
                 except AccessError:
                     pass
 
+    def _delete_and_notify(self):
+        if self.res_model == 'mail.channel':
+            self.env['bus.bus'].sendone((self._cr.dbname, 'mail.channel', self.res_id), {
+                'type': 'mail.attachment_delete',
+                'payload': {
+                    'id': self.id,
+                },
+            })
+        return self.unlink()
```

Comment:
Either specify this is a `ensure_one()` method, or better: adapt code to make it a proper batch method.
In this case I'm suggesting that the payload should be a list of identifying data instead of a single one, and the handler be adapted accordingly.